core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java

```diff
@@ -2444,12 +2449,59 @@ public void setChangeLogStore(ChangeLogStore changeLogStore) {
   @Override
   public long saveBox(Box box) throws IOException {
     if (boxStore != null) {
+      if (box.getId() == AbstractBox.UNSAVED_ID) {
+        try {
+          boxStore.save(box);
+
+          if (autoGenerateIdentificationBarcodes) {
+            box.setIdentificationBarcode(box.getName() + "::" + box.getAlias());
+          }
+          namingScheme.generateNameFor(box);
```

Comment:
Need to do this before setting the box's identificationBarcode that makes use of the name.